Allow hiding buttons in WideBar

flowln opened the pull request:
This allows the user to use the right-click menu (previously unused) on the WideBar to toggle buttons from appearing on the bar. This setting applies globally, and is remembered between runs.

This is so that less common options can be hidden by the user, but still be available to the people that use them. :>

The right-click menu on the main portion of the page continues to always show all the available actions, so that less common actions can stop cluttering the sidebar, but still be readily available on the menu.

flowln committed:
refactor: clean up WideBar a bit

Signed-off-by: flow <flowlnlnln@gmail.com>

flowln committed:
feat(WideBar): allow hiding buttons with right-click

Signed-off-by: flow <flowlnlnln@gmail.com>

flowln committed:
feat(WideBar): allow loading/unloading visibility via a byte array

I really wanted to use a QBitArray :c

Signed-off-by: flow <flowlnlnln@gmail.com>

flowln committed:
feat(InstancePages): save/load wide bar visibility settings

Signed-off-by: flow <flowlnlnln@gmail.com>

flowln committed:
fix: reset wide bar settings when the list of actions changes

This prevents changes to the actions to cause non-intuitive issues in
the Wide bar, hiding items that previously weren't hidden.

Signed-off-by: flow <flowlnlnln@gmail.com>

Scrumplex commented on launcher/ui/widgets/WideBar.cpp:
```diff
@@ -23,137 +28,237 @@ private slots:
+    to->setIcon(from->icon());
+    to->setToolTip(from->toolTip());
+}
+
+void WideBar::contextMenuEvent(QContextMenuEvent* event)
+{
```
maybe use [`QWidget::customContextMenuRequested` signal](https://doc.qt.io/qt-6/qwidget.html#customContextMenuRequested) instead? That's what we use for most of the codebase.

Scrumplex reviewed (approved):
Cool

flowln committed:
reafctor(WideBar): connect to signal instead of overriding menu method

This makes stuff more standard and closer to what we do in other places
in the codebase.

Signed-off-by: flow <flowlnlnln@gmail.com>
